### PR TITLE
Read API secret from file to avoid exposing in proc table

### PR DESCRIPTION
**Pull request by toresbe**
To avoid exposing the API server in the world-readable process table, I generalized the function `readPrivateKey(privateKeyPath string)` to `readFileToString(filePath string)`.

The API secret is read from `path.Join(cfg.ConfigDir, "api_secret")` - and the reading mostly follows the same logic as that used to read the private.key file. I don't know if the zero-length check is butter on the flesk, let me know and I'll axe it.

The api_secret file is already put in place by the revised systemd service definition in the previous pull request on this branch, merged into master.

The deployment of this file using ansible has been tested and known working, so now all that remains is finishing up the github action workflow.

### CODE REVIEW

**Review by sechmann — APPROVED** [review]
LGTM :shipit:

File: cmd/gateway-agent/main.go [hunk 1]
```diff
@@ -44,6 +43,7 @@ func init() {
 	flag.Parse()
 	cfg.WireGuardConfigPath = path.Join(cfg.ConfigDir, "wg0.conf")
 	cfg.PrivateKeyPath = path.Join(cfg.ConfigDir, "private.key")
+	cfg.APIServerPasswordPath = path.Join(cfg.ConfigDir, "api_secret")
```
Comment (jhrv) [inline cmd/gateway-agent/main.go]: suggestion for more consistent naming: `apiserver_password`

**Review by jhrv — APPROVED** [review]
looks good 👌

**Comment by jhrv** [comment]
Fixed dev-mode for gateway-agent, consider adding the creation of the file needed to make this work in the Makefile target `local-gateway-agent`

File: cmd/gateway-agent/main.go (hunk 1 above)
Comment (toresbe) [reply cmd/gateway-agent/main.go]: Good point, will incorporate

**Pull request merged**